lib/rubocop/cop/metrics/parameter_lists.rb

```diff
@@ -15,13 +15,21 @@ def on_args(node)
           count = args_count(node)
           return unless count > max_params
 
+          return if argument_to_set_trace_func?(node)
+
           add_offense(node) do
             self.max = count
           end
         end
 
         private
 
+        def argument_to_set_trace_func?(node)
+          node.parent && node.parent.parent &&
```

Comment:
You could probably use a node matcher, like:

```ruby
def_node_matcher :argument_to_set_trace_func?, <<-PATTERN
  ^^(send nil :set_trace_function _)
PATTERN
```

Each `^` ascends one level in the AST.

Reply:
Thanks, I will try that.